torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -542,7 +542,19 @@ void ProcessGroupAgent::listenLoop() {
       recvWork_ = work;
     }
 
-    if (!rpcRunning_.load() || !work->wait() /* aborted */) {
+    if (!rpcRunning_.load()) {
+      return;
+    }
+    bool workCompleted;
+    try {
+      workCompleted = work->wait();
+    } catch (...) {
```

Comment:
What exceptions could we encounter here? it might be better to handle them explicitly and fail when we encounter an unexpected exception (cc @jjlilley who mentioned something similar on a previous diff)